### ISSUE TITLE
Throwing "Unexpected close" error while registering the account 

### ISSUE THREAD

**Issue by vinodhreddygs** [postalsys/imapapi#76]
Hi @andris9 ,

When I'm trying to register an account on a IMAP API v1.71 version, It's throwing error as "Unexpected close" .
Please refer the below screenshot for the reference:
![image](https://user-images.githubusercontent.com/37541181/126152600-eaa2a547-d011-4022-b108-9d4e30bec12d.png)
Some times we don't get any errors but it shows keep on connecting(i.e. state) 
Please let me know why I'm getting the above error.

Thanks.

**Comment by andris9** [comment]
Can you provide IMAP logs for that account and IMAP settings you use (without the password)? IMAP API logs all IMAP activity console. In general I would guess there is something wrong with IMAP configuration.

**Comment by andris9** [comment]
Additionally you could try to use the the [verifyAccount](https://imapapi.com/api.html#operation/postV1Verifyaccount) API endpoint to see if you get any more specific error response.

**Comment by vinodhreddygs** [comment]
@andris9 when i try to run on **IMAP version 1.6.0**, then it connecting properly.

**Comment by vinodhreddygs** [comment]
> Additionally you could try to use the the [verifyAccount](https://imapapi.com/api.html#operation/postV1Verifyaccount) API endpoint to see if you get any more specific error response.
[logs.log](https://github.com/andris9/imapapi/files/6840481/logs.log)




> Can you provide IMAP logs for that account and IMAP settings you use (without the password)? IMAP API logs all IMAP activity console. In general I would guess there is something wrong with IMAP configuration.

Logs : 
[logs.log](https://github.com/andris9/imapapi/files/6840488/logs.log)

**Comment by andris9** [comment]
The provided logs include information only for the account "qaimap@gmail.com" but not for the errored "orders@pippintec...".

Could you do the following:
1. stop IMAP API
2. clear console output
3. start IMAP API
4. wait until orders@pippin.. fails to connect
5. copy logs

Additionall, do you use any cli flags for IMAP API, eg. `commandTimeout` or `localAddresses`?

**Comment by vinodhreddygs** [comment]
> The provided logs include information only for the account "[qaimap@gmail.com](mailto:qaimap@gmail.com)" but not for the errored "orders@pippintec...".
> 
> Could you do the following:
> 
> 1. stop IMAP API
> 2. clear console output
> 3. start IMAP API
> 4. wait until orders@pippin.. fails to connect
> 5. copy logs
> 
> Additionall, do you use any cli flags for IMAP API, eg. `commandTimeout` or `localAddresses`?

[test-logs.log](https://github.com/andris9/imapapi/files/6840609/test-logs.log)
Please find the logs here. For security purpose just renamed some confidential details.

**Comment by andris9** [comment]
Did you close this issue on purpose? From the logs it seems like IMAP API would be looping connections. It successfully creates a connection and then proceeds until `STATUS INBOX (UIDNEXT MESSAGES HIGHESTMODSEQ UIDVALIDITY)` and then starts over.

Can you provide the same logs for IMAP API 1.6.0 as well for comparison?

**Comment by vinodhreddygs** [comment]
> logs for IMAP API 1.6.0

logs for IMAP API 1.6.0 : 
[test-logs2.log](https://github.com/andris9/imapapi/files/6840801/test-logs2.log)

**Comment by andris9** [comment]
There does not seem to be much difference, only a single connection makes it to IDLE state, otherwise connections seem to end while running the first STATUS command. Are you sure that 1.6.0 works? Can you run IMAP commands against the connected client?

**Comment by vinodhreddygs** [comment]
> There does not seem to be much difference, only a single connection makes it to IDLE state, otherwise connections seem to end while running the first STATUS command. Are you sure that 1.6.0 works? Can you run IMAP commands against the connected client?

yes. Its working as expected.

**Comment by andris9** [comment]
Do you get the logs from the "download logs" section in the web client or straight from the console?

**Comment by andris9** [comment]
If you are using the logs from "download logs" section then it is probably missing some errors, you should get the logs from application stdout/stderr output.

**Comment by andris9** [comment]
> > Do you get the logs from the "download logs" section in the web client or straight from the console?
> 
> got it from "download logs".

Could you get logs from the application output.

1. stop the application and clear console output. if you are running the app as a service, stop the service
2. start the app manually (with your production flags)
```
$ imapapi --whatever-flags-you-use
```
3. Wait until connection you see the connection error
4. copy all log lines from your console

The logs from "donwload logs" include only account related logs but not exceptions.

**Comment by andris9** [comment]
I've been trying to replicate the issue and it seems that the only difference between 1.6.0 and 1.7.1 at the point where connection fails is metrics collection for prometheus, so I guess some kind of exception happens there. Exceptions are not logged in user log (as it is not user level, but app level log entry). So look for any lines in the application output that contains Error in it, eg run this: `imapapi | grep -i Error`

**Comment by andris9** [comment]
Somehow I can't re-create these issues at all 🤷‍♂️

When running with from-env command IMAP API runs the latest version 1.7.1 and connects to provided Redis url correctly:

```
$ from-env npx imapapi --dbs.redis="redis://178.33.49.65:6379/8" --api.port=3005 --api.host="0.0.0.0"
{"level":30,"time":1626763383908,"pid":20489,"hostname":"andrisreinman","msg":"Starting IMAP API","version":"1.7.1"}
{"level":20,"time":1626763383909,"pid":20489,"hostname":"andrisreinman","msg":"IMAP Worker Count","workersImap":4}
{"level":30,"time":1626763384834,"pid":20489,"hostname":"andrisreinman","tid":4,"msg":"Started worker thread"}
{"level":30,"time":1626763384877,"pid":20489,"hostname":"andrisreinman","tid":2,"msg":"Started worker thread"}
{"level":30,"time":1626763384895,"pid":20489,"hostname":"andrisreinman","tid":1,"msg":"Started worker thread"}
{"level":30,"time":1626763384922,"pid":20489,"hostname":"andrisreinman","tid":3,"msg":"Started worker thread"}
{"level":30,"time":1626763385349,"pid":20489,"hostname":"andrisreinman","tid":5,"component":"api","created":1626763385220,"started":1626763385347,"host":"0.0.0.0","port":3005,"protocol":"http","id":"andrisreinman:20489:krbox1fo","uri":"http://0.0.0.0:3005","address":"0.0.0.0","msg":"server started"}
{"level":20,"time":1626763385350,"pid":20489,"hostname":"andrisreinman","tid":5,"msg":"API server started","port":3005,"host":"0.0.0.0","maxSize":5242880}
{"level":30,"time":1626763416884,"pid":20489,"hostname":"andrisreinman","tid":4,"component":"imap-client","account":"pangalink","cid":"10odnetlspg1mde9ul10","sub":"imap-connection","src":"connection","msg":"Established secure TCP connection","cid":"12zur9nrxql0qilqakgc","secure":true,"host":"imap.gmail.com","servername":"imap.gmail.com","port":993,"address":"74.125.205.109","localAddress":"192.168.1.175","localPort":55197,"authorized":true,"algo":"TLS_AES_256_GCM_SHA384","version":"TLSv1.3"}
{"level":20,"time":1626763416913,"pid":20489,"hostname":"andrisreinman","tid":4,"component":"imap-client","account":"pangalink","cid":"10odnetlspg1mde9ul10","sub":"imap-connection","src":"s","msg":"* OK Gimap ready for requests from 90.190.173.90 21mb19832126ltm","cid":"12zur9nrxql0qilqakgc"}
…
```

**Comment by andris9** [comment]
Probably not the issue but maybe you have a corrupted IMAP API version cached in your system. I released v1.7.2, could you try to upgrade and run that version?

**Comment by faradaytrs** [comment]
@andris9 could you release 1.7.2 in docker as well?

**Comment by andris9** [comment]
@faradaytrs It seems that Docker Hub is disabling autobuilds for free accounts so the latest commits were not picked up anymore. I upgraded my plan there and the latest version should be available now.

**Comment by andris9** [comment]
Now that I was checking if the latest Docker build works I think I figured out what actually caused this issue and released a fix (v1.7.3).

IMAP API uses [Bull](https://github.com/OptimalBits/bull) for internal webhook queueing and the way IMAP API provided Redis configuration to Bull did not work anymore for whatever reason (there was a Bull module upgrade between 1.6 and 1.7 though). When connecting an IMAP account IMAP API tries to schedule a webhook about successful authentication and as Bull was not properly set up (there was no local Redis instance to connect to), it probably threw an error and process exited.

Anyway, please try out v1.7.3 and check if the update cleared this issue. If it did not, then re-open the ticket.
